Android/ThirdPartyAuth/app/src/main/java/com/bentley/sample/thirdpartyauth/TokenServerAuthClient.kt

```diff
@@ -0,0 +1,48 @@
+package com.bentley.sample.thirdpartyauth
+
+import com.auth0.android.jwt.JWT
+import com.bentley.itwin.AuthTokenCompletionAction
+import com.bentley.itwin.AuthorizationClient
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import retrofit2.Retrofit
+import java.lang.Exception
+
+class TokenServerAuthClient(tokenServerUrl: String, auth0Token: String) : AuthorizationClient() {
+
+	var auth0Token: String = auth0Token
+
+	val tokenServerUrl: String = tokenServerUrl
+
+	var bentleyToken: String? = null
+
+	private suspend fun fetchBentleyToken(): String {
+		val service = Retrofit.Builder()
+			.baseUrl(tokenServerUrl)
+			.build()
+			.create(TokenService::class.java)
+
+		val response = service.getToken("Bearer ${auth0Token}")
+		return response.string()
+	}
+
+	override fun getAccessToken(tokenAction: AuthTokenCompletionAction?) {
+		var jwt: JWT? = if(bentleyToken == null) null else JWT(bentleyToken!!)
+		if(jwt == null || jwt.isExpired(0)) {
```

Comment:
Space needed between if an (

Reply:
Also, I believe that the leeway should not be set to 0 here. The iOS sample uses a 30 second leeway, I believe.